### PR TITLE
#626: Improve the way to identify the boundary

### PR DESCRIPTION
**Pull request by essobedo**
Fix for #626 

The goal of the PR is to improve the way to identify the boundary pattern especially when we have an empty line before the pattern. If the pattern cannot be found, we get unexpected behavior such as IOException of type "Too many open files in system" because the loop is endless so it creates and opens new files again and again.

### CODE REVIEW

**Comment by davvd** [comment]
@essobedo Let me find a reviewer for this pull request, thanks for submitting it

**Comment by davvd** [comment]
@dmzaytsev please review, thanks

File: src/main/java/org/takes/rq/RqMultipart.java [hunk 1]
```diff
@@ -62,7 +62,7 @@
  * @version $Id$
  * @since 0.9
  */
-@SuppressWarnings("PMD.TooManyMethods")
+@SuppressWarnings({"PMD.TooManyMethods", "PMD.GodClass"})
```
Comment (dmzaytsev) [inline src/main/java/org/takes/rq/RqMultipart.java]: @essobedo please add a new puzzle to refactor this class and remove the new suppression above

File: src/test/java/org/takes/rq/RqMultipartTest.java [hunk 2]
```diff
@@ -507,6 +507,49 @@ public void returnsCorrectPartLength() throws IOException {
     }
 
     /**
+     * RqMultipart.Base can identify the boundary even if the last content to
+     * read before the pattern is an empty line.
+     * @throws IOException If some problem inside
+     */
+    @Test
+    public void identifiesBoundary() throws IOException {
```
Comment (dmzaytsev) [inline src/test/java/org/takes/rq/RqMultipartTest.java]: @essobedo were you be able to reproduce the issue before you fix it?

**Comment by dmzaytsev** [comment]
@essobedo please see couple comments above

File: src/main/java/org/takes/rq/RqMultipart.java (hunk 1 above)
Comment (essobedo) [inline src/main/java/org/takes/rq/RqMultipart.java]: @dmzaytsev no need this is already an issue for this which is #651

Comment (dmzaytsev) [inline src/main/java/org/takes/rq/RqMultipart.java]: @essobedo  okay, no problem

File: src/test/java/org/takes/rq/RqMultipartTest.java (hunk 2 above)
Comment (essobedo) [inline src/test/java/org/takes/rq/RqMultipartTest.java]: @dmzaytsev of course, I only do test-driven development :-)

**Comment by essobedo** [comment]
@dmzaytsev thx for your remarks, please find my answers above

File: src/test/java/org/takes/rq/RqMultipartTest.java (hunk 2 above)
Comment (dmzaytsev) [inline src/test/java/org/takes/rq/RqMultipartTest.java]: @essobedo perfect, thanks :)

**Comment by dmzaytsev** [comment]
@rultor merge

**Comment by essobedo** [comment]
@yegor256 Could you please check? thx in advance

**Comment by essobedo** [comment]
@yegor256 ping

**Comment by essobedo** [comment]
@yegor256 please check the PR, sorry to insist but it is part of my job :-)

**Comment by yegor256** [comment]
@rultor try to merge

**Comment by davvd** [comment]
@elenavolokhova please, review this issue for QA compliance, as per [par.24](http://at.teamed.io/policy.html#24)

**Comment by davvd** [comment]
@rultor deploy now pls

**Comment by elenavolokhova** [comment]
@dmzaytsev 
According to our [quality rules](http://www.teamed.io/qa.html):
 > The code reviewer found at least three problems in the code.
 
 There were no issues found and fixed during review.
Please confirm that you will try to find at least three problems next time.

**Comment by dmzaytsev** [comment]
@elenavolokhova I confitm

**Comment by elenavolokhova** [comment]
@davvd Quality is acceptable here.

**Comment by davvd** [comment]
> @davvd Quality is acceptable here.

@elenavolokhova thank you, let's try to make it "good" next time

**Comment by davvd** [comment]
@dmzaytsev 10 mins was added to the account of @elenavolokhova (for QA review), in transaction 84061934; Thank you, I have added **22 mins** to your account in payment/transaction "`84061962`", time consumed: 123 hours and 16 mins; review comments (c=7) added as a bonus; +22 to your rating, your total score is [+2120](http://www.netbout.com/b/35166?open=rating)

**Pull request merged**